general/array.hpp

```diff
@@ -205,7 +209,10 @@ class Array
    inline void Copy(Array &copy) const;
 
    /// Make this Array a reference to a pointer.
-   inline void MakeRef(T *, int);
+   /** When @a own_data is true, the pointer @a data_ must be allocated with
+       MemoryType given by MemoryManager::GetHostMemoryType(). */
+   inline void MakeRef(T *data_, int size_, bool own_data = false,
+                       MemoryType mt = MemoryManager::GetHostMemoryType());
```

Comment:
Since the two new parameters are being added in this PR, it maybe better, for consistency, to use the same ordering as in `Memory::Wrap`: https://github.com/mfem/mfem/blob/756fd52c2b078acd4c64e235b5fb24093fb30d83/general/mem_manager.hpp#L261

What do you think @pazner?

We just need to update the places where more than 2 parameters are used -- these should be easy to track -- all of them should be in this PR and should result in compilation error if the arguments are switched just here in the declaration and definition.

Reply:
My only reservation is that if you want to explicitly pass `own = false`, then you need to also explicitly pass the `MemoryType`, which is not relevant. We could also add a three-argument version with `own` but without `mt`.

Reply:
I think having
```c++
void MakeRef(T *data_, int size_, bool own_data = false);
```
will be consistent with the constructor
```c++
Array(T *data_, int size_, bool own_data = false);
```
which is being extended in this PR from `Array(T *data_, int size_)`.

We should probably also add constructor (similar to the 4-parameter `MakeRef`):
```c++
Array(T *data_, int size_, MemoryType mt, bool own_data);
```
removing the default values for the last two arguments, so there is no ambiguity with the above 3-argument version; for the same reason, we should also remove the default values from the last two arguments of:
```c++
void MakeRef(T *data_, int size_, MemoryType mt, bool own_data);
```
In principle, we can have a default value for `own_data` but I'm not sure there's a natural choice here. In the 3-argument versions, `own_data = false` reproduces the previous behavior when 2 arguments are used.

Reply:
Does 0702739a69ec549d493e0b60cbbd67fe6bdf3633 look OK?

Reply:
Yes, looks good to me.